packages/caravan-wallets/src/bitbox.ts

```diff
@@ -0,0 +1,467 @@
+    });
+  }
+}
+
+/**
+ * Class for wallet extended public key (xpub) interaction at a given BIP32 path.
+ */
+export class BitBoxExportExtendedPublicKey extends BitBoxInteraction {
+  bip32Path: string;
+
+  network: BitcoinNetwork;
+
+  includeXFP: boolean;
+
+  /**
+   * @param {string} bip32Path path
+   * @param {string} network bitcoin network
+   * @param {boolean} includeXFP - return xpub with root fingerprint concatenated
+   */
+  constructor({ bip32Path, network, includeXFP }) {
+    super();
+    this.bip32Path = bip32Path;
+    this.network = network;
+    this.includeXFP = includeXFP;
+  }
+
+  messages() {
+    return super.messages();
+  }
+
+  /**
+   * Retrieve extended public key (xpub) from BitBox device for a given BIP32 path
+   * @example
+   * import {BitBoxExportExtendedPublicKey} from "@caravan/wallets";
+   * const interaction = new BitBoxExportExtendedPublicKey({network, bip32Path});
+   * const xpub = await interaction.run();
+   * console.log(xpub);
+   */
+  async run() {
+    return await this.withDevice(async (pairedBitBox) => {
+      const xpub = await pairedBitBox.btcXpub(
+        convertNetwork(this.network),
+        this.bip32Path,
+        this.network === 'mainnet' ? 'xpub' : 'tpub',
+        false);
+      if (this.includeXFP) {
+        const rootFingerprint = await pairedBitBox.rootFingerprint();
+        return { xpub, rootFingerprint };
+      }
+      return xpub;
+    });
+  }
+}
+
+export class BitBoxRegisterWalletPolicy extends BitBoxInteraction {
+  walletConfig: MultisigWalletConfig;
+
+  constructor({
+    walletConfig
+  }: { walletConfig: MultisigWalletConfig}) {
+    super();
+    this.walletConfig = walletConfig;
+  }
+
+  messages() {
+    const messages = super.messages();
+    return messages;
+  }
+
+  async run() {
+    return await this.withDevice(async (pairedBitBox) => {
+      await this.maybeRegisterMultisig(pairedBitBox, this.walletConfig);
+      // BitBox does not use HMACs for registrations, so we we don't return anything here.
+    });
+  }
+}
+
+export class BitBoxConfirmMultisigAddress extends BitBoxInteraction {
+  network: BitcoinNetwork;
+
+  bip32Path: string;
+
+  walletConfig: MultisigWalletConfig;
+
+  constructor({ network, bip32Path, walletConfig }) {
+    super();
+    this.network = network;
+    this.bip32Path = bip32Path;
+    this.walletConfig = walletConfig;
+  }
+
+  /**
+   * Adds messages about BIP32 path warnings.
+   */
+  messages() {
+    const messages = super.messages();
+    return messages;
+  }
+
+  async run() {
+    const display = true;
+    return await this.withDevice(async (pairedBitBox) => {
+      const { scriptConfig } = await this.maybeRegisterMultisig(pairedBitBox, this.walletConfig);
+      const address = await pairedBitBox.btcAddress(
+        convertNetwork(this.network),
+        this.bip32Path,
+        scriptConfig,
+        display,
+      );
+      return {
+        address,
+        serializedPath: this.bip32Path,
+      };
```

Comment:
Looks like this is mostly working—noticing we get back a testnet address when on regtest
<img width="1707" alt="image" src="https://github.com/user-attachments/assets/49867e6f-2f3f-45a3-a31c-ff78a480fb74">

Reply:
The BitBox02 itself does not actually support regtest right now. Adding support for it would be easy, and in fact I made a PR for it now (https://github.com/BitBoxSwiss/bitbox02-firmware/pull/1302). It would take a little while until it is released and can be used here though.

Another solution here would be to convert the address to the regtest format inside Caravan after the BitBox02 returns it in the testnet format.

What do you think is the best course of action?

Reply:
In favor for keeping it _elegant_ without the regtest hack for P2WSH! Nested segwit works perfectly and that's likely sufficient for now. If your firmware PR makes it in, it will puzzle in nicely.

Thank you sharing and for suggesting the address conversion, it does work well
```
import { decodeSegwitAddress, encodeSegwitAddress } from "./vendor/bech32";
if (this.walletConfig.network == "regtest") {
  if (address.startsWith("tb1q")) {
    try {
      const decoded = decodeSegwitAddress("tb", address);
      if (!decoded) {
        throw new Error("Could not decode address");
      }
      const resp = encodeSegwitAddress(
        "bcrt",
        decoded.version,
        decoded.program
      );
      return {
        address: resp,
        serializedPath: this.bip32Path,
      };
    } catch (err) {
      console.log("error is", err);
    }
  }
}
```

Reply:
@benma any sense of when the updated firmware might be released?

Reply:
It was released a few weeks ago - if you update your BitBox02 to v9.21.0 it will show the proper regtest address prefix. 

You can update to this firmware most easily via BitBoxApp v4.45+.

I also added a commit here to make use of it.